src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -505,6 +528,76 @@ object HttpGoogleServicesDAO extends GoogleServicesDAO with FireCloudRequestBuil
     }
   }
 
+  override def checkGoogleGroupExists(groupEmail: String): Boolean = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    val checkGroupRequest = directoryService.groups.get(groupEmail)
+    Try(executeGoogleRequest(checkGroupRequest)) match {
+      case Failure(_) => false
+      case Success(_) => true
+    }
+  }
+
+  override def deleteGoogleGroup(groupEmail: String): Unit = {
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+    val deleteGroupRequest = directoryService.groups.delete(groupEmail)
+    Try(executeGoogleRequest(deleteGroupRequest)) match {
+      case Failure(_) => {
+        logger.warn(s"Failed to delete group $groupEmail")
+      }
+      case Success(_) => {}
+    }
+  }
+
+  /**
+    * create a new Google group
+    * @param groupEmail     the name of the new Google group to be created. it must be formatted as an email address
+    * @return               Option of the groupEmail address if all above is successful, otherwise if the group creation
+    *                        step failed, return Option of an empty string
+    */
+  override def createGoogleGroup(groupEmail: String): Option[String] = {
+    val newGroup = new Group().setEmail(groupEmail)
+    val directoryService = getDirectoryManager(getDelegatedCredentialForAdminUser)
+
+    val insertRequest = directoryService.groups.insert(newGroup)
+    val setEmail = Try(executeGoogleRequest[Group](insertRequest)) match {
+      case Failure(_) => {
+        logger.warn(s"Could not create new group called $groupEmail")
+        "" // return empty string
```

Comment:
A more idiomatic approach is to use `Option.empty` which allows for easier case matching for callers.

Reply:
much cleaner, thank you!